analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/CompressedStreamHandler.kt

```diff
@@ -0,0 +1,23 @@
+package de.maibornwolff.codecharta.serialization
+
+import java.io.BufferedInputStream
+import java.io.InputStream
+import java.util.zip.GZIPInputStream
+
+object CompressedStreamHandler {
+
+    fun handleInput(input: InputStream): InputStream {
+        var content = input
+        if (!input.markSupported()) content = BufferedInputStream(input)
+        content.mark(2)
+        val byteArray = content.readNBytes(2)
+        content.reset()
+        if (isGzipByteHeader(byteArray)) return GZIPInputStream(content)
+        return content
+    }
```

Comment:
- Maybe find a better name of this function that expresses what the function is actually doing.
- Please always use `{}` for control / if / loop statements (even for one line if-blocks).

Reply:
`handleInput` => `wrapInput`